### PR TITLE
HBASE-22808 HBCK Report showed the offline regions which belong to di…

### PR DESCRIPTION
**Pull request by infraio**
…sabled table

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/CatalogJanitor.java [hunk 1]
```diff
@@ -672,6 +672,11 @@ private void checkServer(byte [] metaTableRow, RegionLocations locations) {
         if (sn == null) {
           continue;
         }
+        // skip the offline regions which belong to disabled table.
+        if (services.getTableStateManager()
+            .isTableState(location.getRegion().getTable(), TableState.State.DISABLED)) {
```
Comment (saintstack) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/CatalogJanitor.java]: Oh. I see. I didn't see it in my reproduction of the issue.

Could be DISABLED or DISABLING. Thats what we check in other places IIRC.

Comment (infraio) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/CatalogJanitor.java]: There already a method which call isTableDisabled. Let me change to use that one.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 140 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 487 | master passed |
| +1 | compile | 75 | master passed |
| +1 | checkstyle | 103 | master passed |
| +1 | shadedjars | 370 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 52 | master passed |
| 0 | spotbugs | 301 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 298 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 418 | the patch passed |
| +1 | compile | 82 | the patch passed |
| +1 | javac | 82 | the patch passed |
| +1 | checkstyle | 103 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 367 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1386 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 45 | the patch passed |
| +1 | findbugs | 342 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 14351 | hbase-server in the patch failed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 19168 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-465/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/465 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f597cb0e13d3 4.15.0-55-generic #60-Ubuntu SMP Tue Jul 2 18:22:20 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-465/out/precommit/personality/provided.sh |
| git revision | master / 547cec4078 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-465/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-465/1/testReport/ |
| Max. process+thread count | 4714 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-465/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 120 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 379 | master passed |
| +1 | compile | 55 | master passed |
| +1 | checkstyle | 75 | master passed |
| +1 | shadedjars | 271 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 39 | master passed |
| 0 | spotbugs | 258 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 250 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 307 | the patch passed |
| +1 | compile | 57 | the patch passed |
| +1 | javac | 57 | the patch passed |
| +1 | checkstyle | 74 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 272 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 964 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 34 | the patch passed |
| +1 | findbugs | 267 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 17883 | hbase-server in the patch failed. |
| +1 | asflicense | 131 | The patch does not generate ASF License warnings. |
| | | 21602 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.namespace.TestNamespaceAuditor |
|   | hadoop.hbase.client.TestSnapshotDFSTemporaryDirectory |
|   | hadoop.hbase.replication.TestReplicationKillSlaveRSWithSeparateOldWALs |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-465/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/465 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux b016921e2d3f 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-465/out/precommit/personality/provided.sh |
| git revision | master / 547cec4078 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-465/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-465/2/testReport/ |
| Max. process+thread count | 4972 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-465/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**